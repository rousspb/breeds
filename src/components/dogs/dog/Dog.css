.dog-card {
  position: relative;
}
.dog-card img {
  height: 150px;
  width: 150px;
  border-radius: 5px;
  position: relative;
  cursor: pointer;
  border: 3px solid #d08f8f;
}

.dog-card img:hover {
  opacity: 0.2;
}

.dog-card.no-hover img:hover {
  opacity: 1;
}

.dog-card.no-hover img {
  cursor: default;
}

.show-me {
  position: absolute;
  top: 40%;
  left: 0;
  right: 0;
  z-index: -1;
  color: deeppink;
}

.pink {
  color: deeppink;
}

.dog-details-header {
  margin-bottom: 30px;
}

.cucum {
  text-align: left;
  margin-left: 80px;
  display: block;
}