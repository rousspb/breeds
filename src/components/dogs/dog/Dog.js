import React, { Component} from 'react';
import { Link } from "react-router-dom";
import axios from 'axios';
import './Dog.css'

class Dog extends Component {
  state = {
    breeds: [],
    isHover: false
  }
  componentDidMount() {
    axios.get(`https://dog.ceo/api/breed/${this.props.dogName}/images/random/1`)
    .then(res => {
      const breeds = res.data.message;
      this.setState({breeds});
    })
    .catch(error => {
      console.log(error);
    });
  }

  onMouseEnterHandler = () => {
    this.setState({ isHover: true})
  }

  onMouseLeaveHandler = () => {
    this.setState({ isHover: false })
  }
  render() {
    return (
      <Link to={`/dog/${this.props.dogName}`}>
        <div className="dog-card">
          <img src={this.state.breeds[0]} onMouseOver={this.onMouseEnterHandler} onMouseLeave={this.onMouseLeaveHandler} alt="Dog"/>
          {this.state.isHover ? <div className="show-me">Show me more!</div> : ''}
        </div>
      </Link> 
    )
  }
}
export default Dog;
