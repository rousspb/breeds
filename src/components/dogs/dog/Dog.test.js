import React from 'react';
import renderer from 'react-test-renderer';
import Dog from './Dog';
import mockAxios from 'axios';

jest.mock('react-router-dom', () => ({ Link: 'Link' }));
describe('Dog Card', () => {
  let component;
  
  mockAxios.get.mockImplementationOnce(() =>
    Promise.resolve({
      data: { message: ['Bob'] }
    })
  );

  function createComponent() {
    return renderer.create(<Dog dogName={'Bob'} key="1"/>);
  }

  it('should render a snapshot of Dog', () => {
    component = createComponent();
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a dog card', () => {
    component = createComponent();
    const instance = component.getInstance();
    expect(instance.state.isHover).toEqual(false);
  });
})