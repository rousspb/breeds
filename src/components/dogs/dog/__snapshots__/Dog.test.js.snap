// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dog Card should render a snapshot of Dog 1`] = `
<Link
  to="/dog/Bob"
>
  <div
    className="dog-card"
  >
    <img
      alt="Dog"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
    />
    
  </div>
</Link>
`;
